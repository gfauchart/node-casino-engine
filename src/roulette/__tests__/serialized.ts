import RouletteEngine, { SlotBetType } from '../index';

const playerId = '112233';

test('Roulette Engine serialize should remenber bets', () => {
  const engine = new RouletteEngine();

  const serialized = engine
    .registerPlayer({playerId, balance: 110 })
    .placeBet(playerId, {type : SlotBetType.Straight, value: 2, amount: 2})
    .serialize();

  const bets = new RouletteEngine(serialized).getBets()

  expect(bets).toHaveLength(1);
  expect(bets).toMatchSnapshot();
});

test('Roulette Engine serialize should remenber players', () => {
  const engine = new RouletteEngine();

  const serialized = engine
    .registerPlayer({playerId, balance: 110 })
    .placeBet(playerId, {type : SlotBetType.Straight, value: 2, amount: 2})
    .serialize();

  const players = new RouletteEngine(serialized).getPlayers()

  expect(players).toHaveLength(1);
  expect(players).toMatchSnapshot();
});
