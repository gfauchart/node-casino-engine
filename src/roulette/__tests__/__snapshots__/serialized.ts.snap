// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roulette Engine serialize should remenber bets 1`] = `
Array [
  Object {
    "amount": 2,
    "multiplier": 35,
    "playerId": "112233",
    "winningSlots": Array [
      Object {
        "color": 1,
        "number": 2,
      },
    ],
  },
]
`;

exports[`Roulette Engine serialize should remenber players 1`] = `
Array [
  Object {
    "balance": 108,
    "playerId": "112233",
  },
]
`;
